sklearn/manifold/tests/test_graph_spectral_embedding.py

```diff
@@ -0,0 +1,182 @@
+        labels_er[100:] = 1
+
+        X_sbm = embed_sbm.fit_transform(sbm_sample)
+        X_er = embed_er.fit_transform(er)
+
+        if directed:
+            self.assertEqual(X_sbm.shape, (verts, 2 * communities))
+            self.assertEqual(X_er.shape, (verts, 2 * communities))
+        else:
+            self.assertEqual(X_sbm.shape, (verts, communities))
+            self.assertEqual(X_er.shape, (verts, communities))
+
+        aris = _kmeans_comparison((X_sbm, X_er), (labels_sbm, labels_er), communities)
+        sbm_wins = sbm_wins + (aris[0] > aris[1])
+        er_wins = er_wins + (aris[0] < aris[1])
+
+    self.assertTrue(sbm_wins > er_wins)
+
+
+def test_input_params():
+    X = make_er_graph(10, 20)
+
+    # value error, check n_components type int
+    _test_inputs(X, ValueError, n_components='not_int')
+
+    # n_components must be <= min(X.shape)
+    _test_inputs(X, ValueError, n_components=15)
+
+    # value error, check n_elbow type int
+    _test_inputs(X, ValueError, n_elbows='not_int')
+
+    # value error, check n_elbow > 1
+    _test_inputs(X, ValueError, n_elbows=0)
+
+    # check algorithm string
+    _test_inputs(X, ValueError, algorithm='wrong')
+    _test_inputs(X, TypeError, algorithm=1)
+
+    # svd_solver string
+    _test_inputs(X, ValueError, svd_solver='wrong')
+
+    # regularizer not int, float, or bool
+    _test_inputs(X, TypeError, algorithm='lse', regularizer='wrong')
+    # regularizer not greater than 0
+    _test_inputs(X, ValueError, algorithm='lse', regularizer=-1)
+
+class TestAdjacencySpectralEmbed(unittest.TestCase):
+
+    def test_output_dim(self):
+        _test_output_dim(self, 'ASE')
+
+    def test_output_dim_directed(self):
+        _test_output_dim_directed(self, 'ASE')
+
+    def test_unconnected_warning(self):
+        A = make_er_graph(100, 10)
+        with self.assertWarns(UserWarning):
+            ase = GraphSpectralEmbedding(algorithm='ASE')
+            ase.fit(A)
+        A = make_er_graph(100, 10, directed=True)
+        with self.assertWarns(UserWarning):
+            ase = GraphSpectralEmbedding(algorithm='ASE')
+            ase.fit(A)
+
+    def test_sbm_er_binary_undirected(self):
+        P = np.array([[0.8, 0.2], [0.2, 0.8]])
+        _test_sbm_er_binary(self, 'ASE', P, directed=False)
+
+    def test_sbm_er_binary_directed(self):
+        P = np.array([[0.8, 0.2], [0.2, 0.8]])
+        _test_sbm_er_binary(self, 'ASE', P, directed=True)
+
+
+class TestLaplacianSpectralEmbed(unittest.TestCase):
+    def test_output_dim(self):
+        _test_output_dim(self, 'LSE')
+
+    def test_output_dim_directed(self):
+        _test_output_dim_directed(self, 'LSE')
+
+    def test_sbm_er_binary_undirected(self):
+        P = np.array([[0.8, 0.2], [0.2, 0.8]])
+        _test_sbm_er_binary(self, 'LSE', P, directed=False)
+
+    def test_sbm_er_binary_directed(self):
+        P = np.array([[0.8, 0.2], [0.2, 0.8]])
+        _test_sbm_er_binary(self, 'LSE', P, directed=True)
```

Comment:
As we aren't relying on `unittest`, could you change those test to use `pytest` please? 🙂 

Some of the `_test*` can be made `test*` and parametrised using pytest. For instance:

```python
@pytest.mark.parametrize("method", ["LSE", "ASE"])
@pytest.mark.parametrize("P", [np.array([[0.8, 0.2], [0.2, 0.8]])])
@pytest.mark.parametrize("directed", [True, False])
def test_sbm_er_binary(method, P, directed=False, sparse=True, *args, **kwargs):
    # ...
```